sound/soc/sof/ipc4-topology.c

```diff
@@ -1987,7 +2003,7 @@ static int sof_ipc4_prepare_src_module(struct snd_sof_widget *swidget,
 	 * SRC does not perform format conversion, so the output channels and valid bit depth must
 	 * be the same as that of the input.
 	 */
-	in_audio_fmt = &available_fmt->input_pin_fmts[input_format_index].audio_fmt;
+	in_audio_fmt = &src->base_config.audio_fmt;
```

Comment:
not able to review this one, not sure how this is about handling repeated calls when already prepared.